### ISSUE TITLE
Random Unexpected Death

### ISSUE THREAD

**Issue by aurakami** [HabitRPG/habitica#2630]
Hey all! My partner (who doesn't have an account on here, thus me posting) just randomly died. It stated he took 17.4 damage (though he may or may not have had more health left than that) and while our party is currently in a quest, there was not enough damage (1.4) to warrant his death. He has no dailies to detract from his health as well, just habits and to-dos.

As a side note, is it possible to give him certain armor pieces back? He worked really hard to get all the special holiday outfits of each class and dying caused him to lose some of those pieces. (Though the death message stated that his Yetti Tamer Robe was "broken" however it still seems to be there.)

To quote him:

"Snowflake Shield, Ski-sassin Parka (whatever the head/helmet of the ski-sassin set is).

I also have a bunch of garbled habits added to my habit, all green colored.

I also have To Do's that I did a while ago, reappeared in my list of To-Do's."

I have attached a few screenshots below. If more information is needed, let us know.

![alex_habit_death](https://f.cloud.github.com/assets/4622000/2026820/58fcb6c8-889f-11e3-89f0-5d00da36f6b3.png)

![habitrpgrandomdeath_randomhabits](https://f.cloud.github.com/assets/4622000/2026834/57e73cbc-88a0-11e3-844f-b2da38212f64.png)

![habitrpgrandomdeath_duplicatehabits](https://f.cloud.github.com/assets/4622000/2026836/629ed2a0-88a0-11e3-9c3e-98465a6eb403.png)

Thank you!

**Comment by mentalisttraceur** [comment]
Just registered an account on GitHub (surprising I didn't already get around to it, given that I am involved in software development with a couple of open source projects...), I'm the person whom the above bug report concerns.

Further details: We were having internet issues today, including general lagginess and slowness, and at least once over the last couple of days I had a glitch where things seemed to get double counted, with the second count being seemingly delayed by a while. I didn't bother to debug because I am fairly busy and it wasn't significant enough to effect my gameplay...

Anyway, earlier today I had gotten the Snowflake shield, completing not just the Healer limited edition equipment, but also finishing getting them for every class. (I was saving Healer for last since that's the class I wanted to stay at long-term.) I equipped the Healer gear, but then later when my partner (aurakami, the reporter of this issue) was on HabitRPG from her account, she did not see my outfit update from the previous (I was wearing the Candycane Mage holiday edition set previously), even after several reloads on her age (and no, I do not have Use Costumes turned on). So I thought, maybe there was some hiccup with stuff updating from my browser, so I went and reloaded my HabitRPG page. ..a-a-a-and that's when I got hit by the "You died" thing, out of no-where. I'm almost dead certain it said I took 17.4 damage (I had at least that much life, possibly more, I don't remember). There were also other notifications under the -17.4 Health one, but I don't remember what they were. I was too stunned to think to take a screenshot before the notifications faded, but I did think to take it right after. Which is the first screenshot you have in the above report. I opened another tab and also loaded up habitRPG, hoping maybe it was a really messed up glitch but that it didn't "stick". No joy, it loaded to the same you died message. So I clicked through it in that new tab, and it proceeded to take away all of my gold/silver, add 50 health, and claim that my Yeti Tamer Robe broke. Checking inside my inventory/equipment however, I see that I still have the Yeti Tamer Robe (I dare not click on it though, in case more glitchy-ness deprives me of it or something else), but that the Snowflake Shield and the Ski-sassin Hood (or whatever the helmet one is called - I just looked up 'parka' and I realize now that the head piece is not called the parka, but I don't recall what it is and I don't want to spend another 3 gems on a class switch just to get back something that shouldn't have been taken away in the first place) are both missing. The Snowflake Shield still shows up on my avatar as if equipped but is not in my inventory (in the second screenshot above you can see that it still looks as if my avatar has it equipped but that it has re-appeared in my Rewards, suggesting I no longer have it.

My suspicion is that somewhere along the line, some events (me hitting minuses and pluses on my habits and getting some todo's done) go counted once, but also got queued up, and that a page reload somehow brought the queued up stale instances of the events out, and ran the effects for those again.

**Comment by mentalisttraceur** [comment]
Further information: in the day or so leading up to the above situation, I was also having (just as mentioned in the referenced issue #2663) occurrences of issue #2554. I don't know if that helps you guys pinpoint the issue any.

Also question: I see in the first referenced issue's comment that it says special edition items will always be re-buyable if they were bought during the special edition period. That's great. Just to confirm though, that applies to special edition items from another class right? (i.e. I lost both a Snowflake shield and a Ski-sassin hat/hood/whatever, and am currently a healer. If I remain a healer until after the 31st, then switch back to a rogue sometime later, I will still be able to re-buy the Ski-sassin head gear, correct?)

**Comment by mentalisttraceur** [comment]
Possibly related, I also just noticed that all of my stats on my habits got lost - I just deleted all of those garbled automatically added habits, so I don't know if maybe that data got moved into those habits, but yeah. The only stats seem to be two data points from 2013-12-27, which is I think when I first started using HabirRPG consistently and set up all of those habits, and one's from today or so. (Except Habits I added very recently, like on the 30th, so after the random death, still have their stats. Ones I added recently but before the random death, Habit garbling, etc, have no stats on them.)
